File: 2. python/3. array-list/lecture 1/singlenumber.py
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        n = len(nums)
        
        for i in range(n):
            count = 0
            
            for j in range(n):
                if nums[j] == nums[i]:
                    count += 1
            
            if count == 1:
                return nums[i]
        
        return -1

if __name__ == "__main__":
    sol = Solution()
    nums = [3,3,4, 1, 2, 1, 2]
    print(f"Single Number: {sol.singleNumber(nums)}")

File: 1. cpp/3. array/lecture 1/single number.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++) 
        {
            int count = 0;
            for (int j = 0; j < nums.size(); j++) 
            {
                if (nums[j] == nums[i]) {
                    count++;
                }
            }
            if (count == 1) {
                return nums[i];
            }
        }
        return -1; // Should not reach here
    }
};

int main() {
    Solution sol;
    vector<int> nums = {1, 1, 2, 3, 2};
    cout << "Single Number: " << sol.singleNumber(nums) << endl;
    return 0;
}